Makefile: Add kind-image target

joestringer opened the pull request:
Add a new kind-image target, mirroring the microk8s target for building
and tagging the image.

To use:

    $ make kind
    <install Cilium>
    $ make kind-image
    $ kubectl -n kube-system set image daemonset/cilium cilium-agent=localhost:5000/cilium/cilium-dev:local

joestringer committed:
Makefile: Add kind-image target

Add a new kind-image target, mirroring the microk8s target for building
and tagging the image.

To use:

$ make kind
<install Cilium>
$ make kind-image
$ kubectl -n kube-system set image daemonset/cilium cilium-agent=localhost:5000/cilium/cilium-dev:local

Signed-off-by: Joe Stringer <joe@cilium.io>

glibsm commented on Makefile:
```diff
@@ -488,16 +488,25 @@ check-microk8s: ## Validate if microk8s is ready to install cilium.
+microk8s: check-microk8s ## Build cilium-dev docker image and import to microk8s
 	$(QUIET)$(MAKE) dev-docker-image DOCKER_IMAGE_TAG=$(LOCAL_IMAGE_TAG)
 	@echo "  DEPLOY image to microk8s ($(LOCAL_IMAGE))"
-	$(QUIET)$(CONTAINER_ENGINE) tag $(IMAGE_REPOSITORY)/cilium-dev:$(LOCAL_IMAGE_TAG) $(LOCAL_IMAGE)
```
this change doesn't seem related to the commit message

joestringer replied:
Yeah arguably it should be two commits, rework the microk8s target and add the kind target. When we move the LOCAL_IMAGE declaration out of the top-level of the Makefile into the target-specific variable, it means we no longer need this step.

joestringer replied:
^^ Split out for better clarity.

joestringer committed:
Makefile: Use target-specific vars for microk8s

The microk8s target previously relied on an env var declared at the
top-level of the Makefile, which will conflict with an upcoming commit.
Move it into a target-specific variable. This also means we can remove
the redundant tagging operation.

Signed-off-by: Joe Stringer <joe@cilium.io>
